packages/codemod/src/transform/worker.ts

```diff
@@ -2,17 +2,21 @@ import { workerData, parentPort } from 'node:worker_threads';
 import { readFile, writeFile } from 'node:fs/promises';
 import { createTwoFilesPatch } from 'diff';
 import picocolors from 'picocolors';
+import type { JobWorkerData } from './runner.js';
+import type { Transform } from '../utils/types.js';
 
-const { transformerPath, jobs, options } = workerData;
+const { transformerPath, jobs, options } = workerData as JobWorkerData;
 
-const fileTransformer = (await import(transformerPath)).transform;
+const { transform } = (await import(transformerPath)) as {
+  transform: Transform;
+};
```

Comment:
Casting isn't ideal but it's the simplest way to at least get some type information into the worker script.